src/nvim/quickfix.c

```diff
@@ -1583,116 +1622,134 @@ static qf_info_T *ll_get_or_alloc_list(win_T *wp)
   return wp->w_llist;
 }
 
-/*
- * Copy the location list from window "from" to window "to".
- */
-void copy_loclist(win_T *from, win_T *to)
+// Copy location list entries from 'from_qfl' to 'to_qfl'.
+static int copy_loclist_entries(const qf_list_T *restrict from_qfl,
+                                qf_list_T *restrict to_qfl,
+                                qf_info_T *to_qi)
+  FUNC_ATTR_NONNULL_ALL
 {
-  qf_info_T   *qi;
-  int idx;
   int i;
+  const qfline_T *from_qfp;
+
+  // copy all the location entries in this list
+  for (i = 0, from_qfp = from_qfl->qf_start;
+       i < from_qfl->qf_count && from_qfp != NULL;
+       i++, from_qfp = from_qfp->qf_next) {
+    if (qf_add_entry(to_qi,
+                     to_qi->qf_curlist,
+                     NULL,
+                     NULL,
+                     from_qfp->qf_module,
+                     0,
+                     from_qfp->qf_text,
+                     from_qfp->qf_lnum,
+                     from_qfp->qf_col,
+                     from_qfp->qf_viscol,
+                     from_qfp->qf_pattern,
+                     from_qfp->qf_nr,
+                     0,
+                     from_qfp->qf_valid) == FAIL) {
+      return FAIL;
+    }
 
-  /*
-   * When copying from a location list window, copy the referenced
-   * location list. For other windows, copy the location list for
-   * that window.
-   */
-  if (IS_LL_WINDOW(from))
+    // qf_add_entry() will not set the qf_num field, as the
+    // directory and file names are not supplied. So the qf_fnum
+    // field is copied here.
+    qfline_T *const prevp = to_qfl->qf_last;
+    prevp->qf_fnum = from_qfp->qf_fnum;  // file number
+    prevp->qf_type = from_qfp->qf_type;  // error type
+    if (from_qfl->qf_ptr == from_qfp) {
+      to_qfl->qf_ptr = prevp;  // current location
+    }
+  }
+
+  return OK;
+}
+
+// Copy the specified location list 'from_qfl' to 'to_qfl'.
+static int copy_loclist(const qf_list_T *restrict from_qfl,
+                        qf_list_T *restrict to_qfl,
```

Comment:
Does it work after removing `restrct`?

Reply:
will try.

Reply:
yes, test passes, no crash. I removed all `restrict` cases in quickfix.c. Why are we adding that?

Reply:
Optimizations :| . I thought there's no risks if it passed on the ASAN job.

Reply:
Does it pass if `restrict` is kept for primitve types like `char`?

Reply:
> Does it pass if `restrict` is kept for primitve types like `char`?

I would rather just avoid it entirely. Compilers don't really need these explicit hints, and it adds a lot of work for us to mentally verify these hints.